### PR TITLE
GPJax Frontend

### PR DESCRIPTION
**Pull request by thomaspinder**
This PR introduces a frontend for GPJax. There is an accompanying notebook for this. 

Similar to GPFlow, a frontend is established by creating a wrapper around the base kernel class in GPJax. Any model within GPJax will therefore be supported.

Associated with issue #74

### CODE REVIEW

**Comment by thomaspinder** [comment]
I can see that a test has failed for Python 3.8 and 3.9 as there's no function `B.from_numpy()`. I ran the notebook locally on 3.9 with no issues. Do you know if there's an easy fix for this?

File: geometric_kernels/frontends/jax/gpjax.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+            y (Float[Array, "M D"]): An M x D matrix of inputs.
+
+        Returns:
+            Float[Array, "N M"]: The N x M covariance matrix.
+        """
+        matrix = jnp.asarray(self.kernel_fn(params, x, y))
+        return matrix
+
+
+class GeometricKernel(gpjax.kernels.AbstractKernel):
```
Comment (stoprightthere) [inline geometric_kernels/frontends/jax/gpjax.py:47]: Suggest renaming it to `GPJaxGeometricKernel`. All our other wrappers follow this convention, like `GPytorchGeometricKernel`.

**Comment by stoprightthere** [comment]
Hey @thomaspinder, thanks for your PR! I'm going to give a full review in a short time. For now, this `from_numpy` problem seems to stem from jax version being different or something like this. We will figure out how to fix it and update our main accordingly.

**Review by stoprightthere — APPROVED** [review]
Very nice PR!

Regarding `jax` failing, this is because `jax` introduced new `Array` type instead of `DeviceArray` since `jax-0.4.1`. I proposed a PR to `lab` to incorporate that change.

You may either wait till `lab` is updated to be up-to-date with `jax-0.4.1` or cap `jax` version in requirements (so that it is older than `0.4.1`)

**Comment by thomaspinder** [comment]
> Very nice PR!
> 
> Regarding `jax` failing, this is because `jax` introduced new `Array` type instead of `DeviceArray` since `jax-0.4.1`. I proposed a PR to `lab` to incorporate that change.
> 
> You may either wait till `lab` is updated to be up-to-date with `jax-0.4.1` or cap `jax` version in requirements (so that it is older than `0.4.1`)

Thanks! I've capped JAX for now.

**Pull request merged**